docs/machine-learning/how-to-guides/consuming-model-ml-net.md

```diff
@@ -1,32 +1,160 @@
+Learn how to save and load trained models in your application. 
 
-When the model metrics look good to you, it's time to 'operationalize' the model. The `model` object you built can be consumed, persisted, and reused in different environments, applying the same steps that it 'learned' during training.
+Throughout the model building process, a model lives in memory and is accessible throughout the application's lifecycle. However, once the application stops running, if the model is not saved somewhere locally or remotely, it's no longer accessible. Typically models are used at some point after training in other applications either for inference or re-training. Therefore, it's important to store the model. Save and load models using the steps described in subsequent sections of this document when using data preparation and model training pipelines like the one detailed below. Although this sample uses a linear regression model, the same process applies to other ML.NET algorithms.
 
-To save the model to a file, and reload it (potentially in a different context), use the following example:
+```csharp
+HousingData[] housingData = new HousingData[]
+            {
+    new HousingData
+    {
+        Size = 600f,
+        HistoricalPrices = new float[] { 100000f ,125000f ,122000f },
+        CurrentPrice = 170000f
+    },
+    new HousingData
+    {
+        Size = 1000f,
+        HistoricalPrices = new float[] { 200000f, 250000f, 230000f },
+        CurrentPrice = 225000f
+    },
+    new HousingData
+    {
+        Size = 1000f,
+        HistoricalPrices = new float[] { 126000f, 130000f, 200000f },
+        CurrentPrice = 195000f
+    }
+};
+
+// Create MLContext
+MLContext mlContext = new MLContext();
+
+// Load Data
+IDataView data = mlContext.Data.LoadFromEnumerable<HousingData>(housingData);
+
+// Define data preparation estimator
+EstimatorChain<RegressionPredictionTransformer<LinearRegressionModelParameters>> pipelineEstimator =
+    mlContext.Transforms.Concatenate("Features", new string[] { "Size", "HistoricalPrices" })
+        .Append(mlContext.Transforms.NormalizeMinMax("Features"))
+        .Append(mlContext.Regression.Trainers.Sdca());
+
+// Train model
+ITransformer trainedModel = pipelineEstimator.Fit(data);
+
+// Save model
+mlContext.Model.Save(trainedModel, data.Schema, "model.zip");
+```
+
+Because most models and data preparation pipelines inherit from the same set of classes, the save and load operations for these components is the same. Depending on your use case, you can either combine the data preparation pipeline and model into a single [`EstimatorChain`](xref:Microsoft.ML.Data.TransformerChain%601) which would output a single [`ITransformer`](xref:Microsoft.ML.ITransformer) or separate them thus creating separate `ITransformers` for each. 
```

Comment:
>save and load operations for these components is the same [](start = 93, length = 57)

maybe be more concrete: the save and load api signature is similar.

Reply:
#resolved